### PR TITLE
feat: 🎸 ProtobufUtil Date <-> Timestamp

### PR DESCRIPTION
**Pull request by ThaddeusJiang**
Date と google/protobuf/Timestamp 自由転換

### CODE REVIEW

File: src/utils/protobuf.util.ts [hunk 1]
```diff
@@ -30,4 +31,38 @@ export class ProtobufUtil {
...
+
+    const seconds = Long.fromNumber(date.getTime() / 1_000);
+    const nanos = (date.getTime() % 1_000) * 1_000_000;
+    return { seconds, nanos };
+  }
+
+  /**
+   * Timestamp -> Date
+   *
+   * memo: Don't modify this function, just copy from https://gitlab.com/plugo-inc/plugo-proto/-/blob/develop/dist/ts-server/ocpp/type.ts#L1704-1714
```
Comment (Ko1103) [inline src/utils/protobuf.util.ts]: Remove this because this is private repository.

File: src/utils/protobuf.util.ts [hunk 2]
```diff
@@ -30,4 +31,38 @@ export class ProtobufUtil {
   public static getBigIntFromInt64(id: Long) {
     return BigInt(id.toString());
   }
+
+  /**
+   * Date -> Timestamp
+   *
+   * memo: Don't modify this function, just copy from https://gitlab.com/plugo-inc/plugo-proto/-/blob/develop/dist/ts-server/ocpp/type.ts#L1692-1702
```
Comment (Ko1103) [inline src/utils/protobuf.util.ts]: Remove this because this is private repository.

File: src/utils/protobuf.util.ts [hunk 3]
```diff
@@ -30,4 +31,38 @@ export class ProtobufUtil {
...
+   * memo: Don't modify this function, just copy from https://gitlab.com/plugo-inc/plugo-proto/-/blob/develop/dist/ts-server/ocpp/type.ts#L1704-1714
+   *
+   * @param t Timestamp
+   */
+  public static timestampToDate(t: Timestamp): Date;
+  public static timestampToDate(t: Timestamp | undefined): Date | undefined;
+  public static timestampToDate(t: Timestamp | undefined): Date | undefined {
+    if (!t) return undefined;
+
+    let millis = (t.seconds.toNumber() || 0) * 1_000;
```
Comment (Ko1103) [inline src/utils/protobuf.util.ts:60]: Is this work? 
0 / 1_000

Comment (ThaddeusJiang) [reply src/utils/protobuf.util.ts:60]: @Ko1103 Yes

  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar

image

Comment (ThaddeusJiang) [reply src/utils/protobuf.util.ts:60]: @Ko1103 You can try it in browser console

image

Comment (Ko1103) [reply src/utils/protobuf.util.ts:60]: OK

**Review by Ko1103 — APPROVED** [review]

**Pull request merged**